src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/26.diff.sql

```diff
@@ -0,0 +1,281 @@
+	FROM dbo.DateTimeSearchParam
+)
+DELETE FROM CTE WHERE row_num > 1
+GO
+
+-- Create nonclustered primary key on the set of non-nullable columns that makes it unique
+IF NOT EXISTS (
+    SELECT * 
+	FROM sys.key_constraints 
+	WHERE name='PK_DateTimeSearchParam' AND type='PK')
+BEGIN
+    EXEC dbo.LogSchemaMigrationProgress 'Adding PK_DateTimeSearchParam'
+	ALTER TABLE dbo.DateTimeSearchParam 
+	ADD CONSTRAINT PK_DateTimeSearchParam PRIMARY KEY NONCLUSTERED(ResourceTypeId, ResourceSurrogateId, SearchParamId, StartDateTime, EndDateTime)
+	WITH (DATA_COMPRESSION = PAGE, ONLINE=ON)
+	ON PartitionScheme_ResourceTypeId(ResourceTypeId) 
+END
+GO
+
+/************************************************************************
+ Insert an entry with id 0 to dbo.QuantityCode table
+************************************************************************/
+EXEC dbo.LogSchemaMigrationProgress 'Insert an entry with id 0 to dbo.QuantityCode table'
+IF NOT EXISTS (SELECT 1 FROM dbo.QuantityCode WHERE QuantityCodeId = 0 AND value = '')
+BEGIN
+    SET IDENTITY_INSERT dbo.QuantityCode ON;
+
+    Insert INTO dbo.QuantityCode (QuantityCodeId, Value)
+    Values (0, '')
+
+    SET IDENTITY_INSERT dbo.QuantityCode OFF;
+END;
+GO
+
+/***************************************************************
+ Insert an entry with id 0 to dbo.System table
+****************************************************************/
+EXEC dbo.LogSchemaMigrationProgress 'Insert an entry with id 0 to dbo.System table'
+IF NOT EXISTS (SELECT 1 FROM dbo.System WHERE SystemId = 0 AND value = '')
+BEGIN
+    SET IDENTITY_INSERT dbo.System ON;
+
+    Insert INTO dbo.System (SystemId, Value)
+    Values (0, '')
+
+    SET IDENTITY_INSERT dbo.System OFF;
+END;
+GO
+
+/******************************************************************
+  Table - dbo.QuantitySearchParam table
+*******************************************************************/
+
+-- Deleting duplicate rows based on all columns
+EXEC dbo.LogSchemaMigrationProgress 'Deleting redundant rows from dbo.QuantitySearchParam'
+GO
+;WITH CTE AS (
+    SELECT ResourceTypeId, ResourceSurrogateId, SearchParamId, SystemId, QuantityCodeId, SingleValue, LowValue, HighValue, IsHistory, ROW_NUMBER() 
+    OVER (
+		PARTITION BY ResourceTypeId, ResourceSurrogateId, SearchParamId, SystemId, QuantityCodeId, SingleValue, LowValue, HighValue, IsHistory
+		ORDER BY ResourceTypeId, ResourceSurrogateId, SearchParamId, SystemId, QuantityCodeId, SingleValue, LowValue, HighValue
+	) row_num
+	FROM dbo.QuantitySearchParam
+)
+DELETE FROM CTE WHERE row_num > 1
+GO
+
+-- Backfill table dbo.QuantitySearchParam with non-null QuantityCodeId value
+EXEC dbo.LogSchemaMigrationProgress 'Back-fill table dbo.QuantitySearchParam'
+UPDATE dbo.QuantitySearchParam
+SET QuantityCodeId = 0 
+WHERE QuantityCodeId IS NULL;
+GO
+
+-- Backfill table dbo.QuantitySearchParam with non-null SystemId value
+EXEC dbo.LogSchemaMigrationProgress 'Back-fill table dbo.QuantitySearchParam'
```

Comment:
nit: can we add different logging statement than above
maybe EXEC dbo.LogSchemaMigrationProgress 'Back-fill column SystemId into the table dbo.QuantitySearchParam'

Reply:
Updated 👍